src/main/resources/db/changesets/20210301_revised_initial_schema.yaml

```diff
@@ -0,0 +1,246 @@
+                references: workspace(workspace_id)
+                foreignKeyName: fk_resource_workspace_id
+                nullable: false
+                deleteCascade: true
+              remarks: |
+                First part of the foreign key to cloud_context. Foreign key will cause
+                resource to be deleted when its associated cloud_context is deleted.
+          - column:
+              name: cloud_platform
+              type: text
+              constraints:
+                nullable: false
+              remarks: |
+                Enum of the cloud: GCP and AZURE are the current types
+                Second part of the foreign key to cloud_context
+          - column:
+              name: resource_id
+              type: text
+              constraints:
+                primaryKey: true
+                nullable: false
+          - column:
+              name: name
+              type: text
+              constraints:
+                nullable: false
+          - column:
+              name: description
+              type: text
+          - column:
+              name: stewardship_type
+              type: text
+              constraints:
+                nullable: false
+              remarks: |
+                Enumeration of how the WSM stewards the resource. Current types are
+                REFERENCE - the resource is a reference to a cloud object not managed by WSM
+                CONTROLLED - the resource is controlled by WSM for lifecycle, IAM, and policy
+          - column:
+              name: resource_type
+              type: text
+              constraints:
+                nullable: false
+              remarks: |
+                Enumeration of the resource type. Resource types are specific to the cloud
+                and resource; e.g., GCS_BUCKET, BIGQUERY_DATASET
+          - column:
+              name: cloning_instructions
+              type: text
+              constraints:
+                nullable: false
+              remarks: |
+                Enumeration of the cloning instructions. Details TBD
+          - column:
+              name: attributes
+              type: jsonb
+              constraints:
+                nullable: true
+              remarks: |
+                Attributes are used by both referenced resources and controlled
+                resources to store resource-specific metadata. This bit of overloading
+                between referenced and controlled resource data allows us to avoid a
+                separate referenced_resource table that would have a single column for its
+                metadata.
+          - column:
+              name: access
+              type: text
+              remarks: |
+                Must be valid, non-null for controlled resources. Must be null for reference resources.
+                Enum of the possible access types of the controlled resource. Values are
+                USER_SHARED, USER_PRIVATE, APP_SHARED, APP_PRIVATE
```

Comment:
why not store the two bits separately in different columns for managedBy and access?

Reply:
I plan to make that change